### ISSUE TITLE
Unable to resolve path to module typescript import 

### ISSUE THREAD

**Issue by TheElegantCoding** [ota-meshi/eslint-plugin-astro#284]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.55.0

### What version of `eslint-plugin-astro` are you using?

0.30.0

### What did you do?

Hi, i have a problem with eslint and unresolved imports in astro files for my normal import resolver i have this configuration in my custom eslint config

```js
settings:
  {
    'import/extensions': [
      '.js',
      '.mjs',
      '.jsx',
      '.ts',
      '.tsx',
      '.d.ts'
    ],
    'import/external-module-folders': [ 'node_modules', 'node_modules/@types' ],
    'import/parsers':
    {
      '@typescript-eslint/parser': [ '.ts', '.tsx', '.d.ts' ]
    },
    'import/resolver':
    {
      node: {
        extensions: [
          '.mjs',
          '.js',
          '.json',
          '.ts',
          '.d.ts'
        ],
        tryExtensions: [
          '.mjs',
          '.cjs',
          '.js',
          '.json',
          '.node',
          '.mts',
          '.cts',
          '.ts',
          '.d.ts'
        ]
      }
    }
```

this is how i handle normal ts and js files for absolute an relative paths, but i am getting this issue in the files

![image](https://github.com/ota-meshi/eslint-plugin-astro/assets/70770072/6c51183d-c8a7-4258-98ad-bf2d00194b57)

i override my eslint config this way but still dont work

```js
  {
    files: [ '*.astro' ],
    parser: 'astro-eslint-parser',
    parserOptions: {
      extraFileExtensions: [ '.astro' ],
      parser: '@typescript-eslint/parser'
    }
  }
```

Also i try to setup astro as a file in the import resolver but doesn't work 

![image](https://github.com/ota-meshi/eslint-plugin-astro/assets/70770072/4663bdf5-4104-491b-9de8-2d04989e2fb3)

this is mu tsxonfig.json

```json
{
  "extends": "astro/tsconfigs/strictest",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@global/*": ["src/global/*"],
      "@module/*": ["src/module/*"],
      "@public/*": ["/public/*"]
    }
}
```

### What did you expect to happen?

Imports should work fine

### What actually happened?

dont work

### Link to Minimal Reproducible Example
here is a repo whit all the configuration is a simple empty astro project you can ignore the postcss and more configurations

do `pnpm i` and then see the `index.astro` file with the error

[repo](https://github.com/TheElegantCoding/import_resolver_astro)

**Comment by TheElegantCoding** [comment]
i update my issue with the reproduction repo

**Comment by TheElegantCoding** [comment]
Maybe this can be relevant too i get this error every time i create a file 
![image](https://github.com/ota-meshi/eslint-plugin-astro/assets/70770072/c49d71d6-88d4-4c0a-b75b-fd1de9626d3a)

**Comment by TheElegantCoding** [comment]
i mange to solved the issue thanks to new error i get, so in the #274 is the solutions i need to add this line in my eslint config

```js
    'import/parsers': {
      'astro-eslint-parser': ['.astro'],
      'espree': ['.js'],
    }
```

it should be good if there is a documentation of this in the readme or the webpage

**Comment by ota-meshi** [comment]
I think you first need to add the eslint-plugin-import configuration for TypeScript. Could you please try it?

https://github.com/import-js/eslint-plugin-import#typescript

**Comment by TheElegantCoding** [comment]
yes that help me, but i still have the issue when i create a new file every time `Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.`

**Comment by ota-meshi** [comment]
Please provide a repository that reproduces the remaining issue.

**Comment by ota-meshi** [comment]
I tried using your repository but cannot reproduce the problem.

**Comment by ota-meshi** [comment]
It looks like you haven't tried the following settings yet. I think you should do that first.

https://github.com/import-js/eslint-plugin-import#typescript

**Comment by markomitranic** [comment]
Happens to me as well, and none of the suggested fixes work. Thank you for the effort in debugging this.
The linter works perfectly fine - but if you create a new `.astro` file, you are met with the following error, until you restart the ESLint server / VSCode.

```
Parsing error: "parserOptions.programs" has been provided for @typescript-eslint/parser.
The file was not found in any of the provided program instance(s): src/components/Client copy.astro
eslint
```

It seems to purely be related to the last segment of the `tseslint.config()` train. If the last argument is removed, this behaviour disappears entirely. In fact it seems to exclusively be related to the languageOptions segment of the last part.

I don't know how eslint works, but something tells me that this means it watches all files that already exist, and if a new file is generated, it can't load its type specs. 🤷 

```ts
import { default as eslint, default as js } from "@eslint/js";
import eslintPluginAstro from "eslint-plugin-astro";
import tseslint from "typescript-eslint";

export default tseslint.config(
  js.configs.recommended,
  eslint.configs.recommended,
  tseslint.configs.recommendedTypeChecked,
  eslintPluginAstro.configs.recommended,
  eslintPluginAstro.configs["jsx-a11y-strict"],
  {
    rules: {
      "require-await": "off",
      "@typescript-eslint/require-await": "error",
      "@typescript-eslint/no-misused-promises": [
        "error",
        { checksVoidReturn: { attributes: false } },
      ],
    },

    /**
     * Some rules require to read type information.
     * @see https://typescript-eslint.io/getting-started/typed-linting
     */
    languageOptions: {
      parserOptions: {
        projectService: true,
        tsconfigRootDir: import.meta.dirname,
      },
    },
  },
);
```

Any thoughts @ota-meshi
